### PR TITLE
Execute sync

### PR DESCRIPTION
**Pull request by amanhigh**

### CODE REVIEW

File: core/src/main/java/com/flipkart/poseidon/datasources/AbstractDataSource.java [hunk 1]
```diff
@@ -50,6 +50,10 @@ public AbstractDataSource(LegoSet legoset, Request request) {
         return this.legoset.getDataSourceExecutor().submit(this.legoset.wrapDataSource(dataSource, dataSource.getRequest()));
     }
 
+    protected DataType executeSync(AbstractDataSource dataSource) throws Exception {
+        return this.execute(dataSource).get();
```
Comment (shrey-garg) [inline core/src/main/java/com/flipkart/poseidon/datasources/AbstractDataSource.java]: lets not spawn an extra thread here, do datasource.call()

File: api/src/main/java/com/flipkart/poseidon/internal/ListDataType.java [hunk 2]
```diff
@@ -16,16 +16,16 @@
...
 
 import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.Collection;
-import java.util.List;
 
 public class ListDataType<T> extends ArrayList<T> implements DataType {
 
+    public ListDataType() {
+    }
```
Comment (shrey-garg) [inline api/src/main/java/com/flipkart/poseidon/internal/ListDataType.java]: lets call super()?

File: api/src/main/java/com/flipkart/poseidon/internal/MapDataType.java [hunk 3]
```diff
@@ -16,16 +16,16 @@
...
 
-import java.util.Arrays;
 import java.util.HashMap;
-import java.util.List;
 import java.util.Map;
 
 public class MapDataType<K, V> extends HashMap<K, V> implements DataType {
 
+    public MapDataType() {
+    }
```
Comment (shrey-garg) [inline api/src/main/java/com/flipkart/poseidon/internal/MapDataType.java]: same

**Comment by shrey-garg** [comment]
This is against master which is 5.x branch, please raise a PR against 4.x too as this is a backward compatible change and Rome is still on 4.x

File: core/src/main/java/com/flipkart/poseidon/datasources/AbstractDataSource.java (hunk 1 above)
Comment (shrey-garg) [inline core/src/main/java/com/flipkart/poseidon/datasources/AbstractDataSource.java]: also call wrapDataSource before doing the .call()

Comment (shrey-garg) [inline core/src/main/java/com/flipkart/poseidon/datasources/AbstractDataSource.java]: True. forgot about that.

Generic type might require a change in Lego too, I'll be taking that up

Comment (amanhigh) [inline core/src/main/java/com/flipkart/poseidon/datasources/AbstractDataSource.java]: Changes done. Completely Forgot .call() was the reason we were doing it :P.

File: api/src/main/java/com/flipkart/poseidon/internal/ListDataType.java (hunk 2 above)
Comment (amanhigh) [inline api/src/main/java/com/flipkart/poseidon/internal/ListDataType.java]: Done

File: api/src/main/java/com/flipkart/poseidon/internal/MapDataType.java (hunk 3 above)
Comment (amanhigh) [inline api/src/main/java/com/flipkart/poseidon/internal/MapDataType.java]: Done

**Comment by pmohankumar** [comment]
Please raise a PR to merge similar changes into 4.x branch.

**Pull request merged**